File: EMS_BACKEND1/src/main/java/com/ems/service/DashboardService.java
package com.ems.service;

import com.ems.dto.AttendanceSummaryDTO;
import com.ems.dto.DashboardOverviewDTO;
import com.ems.dto.LeaveSummaryDTO;
import com.ems.dto.RoleDistributionDTO;

import java.time.LocalDate;

public interface DashboardService {
    DashboardOverviewDTO getDashboardOverview();
    LeaveSummaryDTO getLeaveSummary();
    AttendanceSummaryDTO getAttendanceSummary(LocalDate startDate, LocalDate endDate);
    RoleDistributionDTO getEmployeeRoleDistribution();
}

File: employee-management-frontend/src/services/DashboardService.js
// src/services/dashboardService.js
import axiosInstance from "../api/axiosConfig";

const DASHBOARD_API_URL = "/dashboard";

const DashboardService = {
  getDashboardOverview: () => {
    return axiosInstance.get(`${DASHBOARD_API_URL}/overview`);
  },

  getLeaveSummary: () => {
    return axiosInstance.get(`${DASHBOARD_API_URL}/leave-summary`);
  },

  getAttendanceSummary: (startDate, endDate) => {
    return axiosInstance.get(`${DASHBOARD_API_URL}/attendance-summary`, {
      params: { startDate, endDate },
    });
  },

  getEmployeeRoleDistribution: () => {
    return axiosInstance.get(`${DASHBOARD_API_URL}/role-distribution`);
  },
};

export default DashboardService;